flepimop/gempyor_pkg/src/gempyor/inference_parameter.py

```diff
@@ -246,24 +272,16 @@ def inject_proposal(
             snpi_df (pd.DataFrame): DataFrame for snpi.
 
         Returns:
-            pd.DataFrame, pd.DataFrame: Modified hnpi_df and snpi_df.
+            A tuple of modified DataFrames, the first one for SEIR modifiers and the
+            second for outcome modifiers.
         """
         snpi_df_mod = snpi_df.copy(deep=True)
         hnpi_df_mod = hnpi_df.copy(deep=True)
-
-        # Ideally this should lie in each submodules, e.g NPI.inject, parameter.inject
-
-        for p_idx in range(self.get_dim()):
-            if self.ptypes[p_idx] == "seir_modifiers":
-                snpi_df_mod.loc[
-                    (snpi_df_mod["modifier_name"] == self.pnames[p_idx])
-                    & (snpi_df_mod["subpop"] == self.subpops[p_idx]),
-                    "value",
-                ] = proposal[p_idx]
-            elif self.ptypes[p_idx] == "outcome_modifiers":
-                hnpi_df_mod.loc[
-                    (hnpi_df_mod["modifier_name"] == self.pnames[p_idx])
-                    & (hnpi_df_mod["subpop"] == self.subpops[p_idx]),
-                    "value",
-                ] = proposal[p_idx]
+        for p_idx in range(len(self)):
+            df = snpi_df_mod if self.ptypes[p_idx] == "seir_modifiers" else hnpi_df_mod
```

Comment:
love it - i assume this logic is working fine as pointer-/reference-like updating operation, rather than copy-on-assign to a subsequently ignored `df`?

Reply:
Will add